src/Session.ts

```diff
@@ -1015,22 +1016,62 @@ export class Session extends DebugSession {
         }
     }
 
-    private createSourceMapFileWatcher(sourceMapRoot?: string) {
-        if (this._fileWatcher) {
-            this._fileWatcher.dispose();
-            this._fileWatcher = undefined;
+    private createSourceMapFileWatcher(localRoot: string, sourceMapRoot?: string) {
+        if (this._sourceFileWatcher) {
+            this._sourceFileWatcher.dispose();
+            this._sourceFileWatcher = undefined;
         }
-        if (sourceMapRoot) {
-            this._fileWatcher = workspace.createFileSystemWatcher('**/*.{map}', false, false, false);
-            this._fileWatcher.onDidChange(uri => {
-                this._sourceMaps.reset();
-            });
-            this._fileWatcher.onDidCreate(uri => {
-                this._sourceMaps.reset();
-            });
-            this._fileWatcher.onDidDelete(uri => {
-                this._sourceMaps.reset();
-            });
+
+        const config = workspace.getConfiguration('minecraft-debugger');
+        const reloadOnSourceChangesEnabled = config.get<boolean>('reloadOnSourceChanges.enabled');
+        if (!reloadOnSourceChangesEnabled) {
+            return;
+        }
+
+        const reloadOnSourceChangesDelay = Math.max(config.get<number>('reloadOnSourceChanges.delay') || 0, 0);
```

Comment:
```suggestion
        const reloadOnSourceChangesDelay = Math.max(config.get<number>('reloadOnSourceChanges.delay') ?? 0, 0);
```

since we care about nullish check, not just falsy?